convert2rhel/subscription.py

```diff
@@ -190,6 +190,18 @@ def register_system():
             # <system-name>-release package in one of the steps before
             # RHELC-16
             os_release_file.restore(rollback=False)
+        except (OSError, IOError) as e:
+            loggerinst.critical_no_exit(
+                "Failed to restore the /etc/os-release needed fo subscribing the system with message: %s" % str(e)
+            )
+            raise exceptions.CriticalError(
+                id_="FAILED_TO_SUBSCRIBE_SYSTEM",
+                title="Failed to subscribe system.",
+                description="Failed to restore the /etc/os-release needed fo subscribing the system.",
+                diagnosis="The restore failed with error %s." % (str(e)),
+            )
+
+        try:
```

Comment:
Don't wait for approval to do unit tests, mention what you want to do in chat or elsewhere if you want to double check if it's good. Overall this try catch is fine

Reply:
Yeah, I'll add the unit tests now :)